tuist/XcodeProj
# Fix Tuist generated project (for release)

- Following `RELEASE.md` one of the steps is genreate a tuist project used for the Carthage integration
- The Tuist config was out dated (generationOptions API has changed in Tuist 3.x)
- The steps ordering was incorrect, to ensure a successful generation, the carthage dependency frameworks need to be built first

Test Plan:

- Run `bundle exec rake carthage_update_dependencies`
- Run `tuist generate`
- Verify the project successfully generates
